modules/caddyhttp/reverseproxy/selectionpolicies.go

```diff
@@ -655,12 +655,17 @@ func (s CookieHashSelection) Select(pool UpstreamPool, req *http.Request, w http
 		if err != nil {
 			return upstream
 		}
-		http.SetCookie(w, &http.Cookie{
+		cookie := &http.Cookie{
 			Name:   s.Name,
 			Value:  sha,
 			Path:   "/",
 			Secure: false,
-		})
+		}
+		if req.TLS != nil {
```

Comment:
Can probably also add this check:

```
trusted := caddyhttp.GetVar(req.Context(), caddyhttp.TrustedProxyVarKey).(bool)
xfp, ok, _ := lastHeaderValue(req.Header, "X-Forwarded-Proto")
if req.TLS != nil || (trusted && ok && xfp == "https") {
```

This way it still works as expected if Caddy is behind another proxy which adds the XFP header and Caddy has `trusted_proxies` configured to trust the remote IP.

Reply:
Ahhhh, that makes sense. Thanks.